modules/application/src/main/java/edu/northeastern/truthtree/adapter/basicInfo/BasicInfoDBAdapter.java

```diff
@@ -106,8 +114,8 @@ private List getAllCities() {
    * @return JSONArray representing the data on STATES_URL
    */
   @Override
-  public JSONArray getBasicStatesInfo() {
-    return JSONUtil.readJSONFile(STATES_FILE_PATH);
+  public PageDTO<LocationDTO> getBasicStatesInfo() {
+    return jsonArrayToLocationPage(JSONUtil.readJSONFile(STATES_FILE_PATH));
```

Comment:
You can directly used one-time initialized state values from static block. It will avoid reading from file in every request.

Reply:
addressed